File: Two_Pointers/100.Remove Duplicates from Sorted Array/Solution.java
public class Solution {
    /*
     * @param nums: An ineger array
     * @return: An integer
     */
    public int removeDuplicates(int[] nums) {
        // write your code here
        if (nums == null || nums.length == 0) {
            return 0;
        }

        int j = 0;
        int len = nums.length;
        for (int i = 1; i < len; i++) {
            if (nums[i] != nums[j]) {
                // update value
                j++;
                nums[j] = nums[i];
            }
        }
        return j + 1;
    }
}

File: Two_Pointers/100.Remove Duplicates from Sorted Array/Solution.py
class Solution:
    """
    @param: nums: An ineger array
    @return: An integer
    """
    def removeDuplicates(self, nums):
        # write your code here
        if not nums or len(nums) == 0:
            return 0

        length = len(nums)

        p1, p2 = 0, 1
        while p2 < length:
            if nums[p1] != nums[p2]:
                p1 += 1
                nums[p1] = nums[p2]
                p2 += 1
            else:
                while p2 < length and nums[p2] == nums[p1]:
                    p2 += 1
                if p2 < length:
                    p1 += 1
                    nums[p1] = nums[p2]
                    p2 += 1

        return p1 + 1
